### PR TITLE
Fix poison lock error on proxy app

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

Fix logging string
Fixed poison lock error

## Motivation and Context

The logging string stil reflected the previous name of the proxy application. This is now updated. 
An updated log4rs is also required to log this. 

The proxy app can suffer a poison lock error when the stream handling occurs an error. This fixes the potential error around this. 

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by StriderDM — DISMISSED** [review]

File: applications/tari_merge_mining_proxy/src/main.rs [hunk 1]
```diff
@@ -584,12 +582,10 @@ fn stream_handler(
...
-            let _ = handle_connection(stream, &mut tg, r, &mut kg)
-                .map_err(|e| {
-                    error!(target: LOG_TARGET, "One of the threads crashed: {}", e);
-                    e
-                })
-                .unwrap();
+            let _ = handle_connection(stream, &mut tg, r, &mut kg).map_err(|e| {
+                error!(target: LOG_TARGET, "One of the threads crashed: {}", e);
+                e
+            });
```
Comment (sdbondi) [inline applications/tari_merge_mining_proxy/src/main.rs]: Same here

File: applications/tari_merge_mining_proxy/src/main.rs [hunk 2]
```diff
@@ -515,12 +515,10 @@ fn main() {
...
-    let _ = stream_handler(listener, transient, rules, rt)
-        .map_err(|e| {
-            error!(target: LOG_TARGET, "{}", e);
-            e
-        })
-        .unwrap();
+    let _ = stream_handler(listener, transient, rules, rt).map_err(|e| {
+        error!(target: LOG_TARGET, "{}", e);
+        e
+    });
```
Comment (sdbondi) [inline applications/tari_merge_mining_proxy/src/main.rs]: Surprised clippy didn't complain 
```suggestion
    if let Err(e) = stream_handler(listener, transient, rules, rt) {
        error!(target: LOG_TARGET, "{}", e);
    }
```

**Review by sdbondi — COMMENTED** [review]
LGTM  one nit

**Review by stringhandler — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**